server/ssl/ssl.go

```diff
@@ -8,13 +8,18 @@ import (
 
 // from https://medium.com/@kelseyhightower/optimizing-docker-images-for-static-binaries-b5696e26eb07
 
-var pool *x509.CertPool
-
-func GetRootCAPool() *x509.CertPool {
-	if pool == nil {
-		pool = x509.NewCertPool()
-		pool.AppendCertsFromPEM(pemCerts)
+func CreateCertPool(useSystem bool) (*x509.CertPool, error) {
+	if useSystem {
+		return x509.SystemCertPool()
 	}
+
+	pool := createCertPoolFromEmbedded()
+	return pool, nil
+}
+
+func createCertPoolFromEmbedded() *x509.CertPool {
+	pool := x509.NewCertPool()
+	pool.AppendCertsFromPEM(pemCerts)
```

Comment:
`pool.AppendCertsFromPEM(pemCerts)` returns `true` if the certificates are successfully added. Should we read it's return value and either make `createCertPoolFromEmbedded()` return an error if unsuccessful?

Reply:
Sure. Looks like this was missed by the original authors.

Reply:
Added. FYI: the ok return doesn't return false if it failed. The failures are silent. The false is to say no certs were loaded, which for our usage is close enough.